analyses/ps3_script.py

```diff
@@ -353,4 +353,47 @@ def lorenz_curve(y_true, y_pred, exposure):
     )
 )
 
+# %%
+# PS4 EX2
+import lightgbm as lgb
+# Re-fit the best constrained LGBMRegressor
+best_lgbm = cv_constrained.best_estimator_  # Best model from cross-validation
+eval_results = {}  # Dictionary to store evaluation metrics
+
+# Fit the model with evaluation sets
+best_lgbm.named_steps['lgbm'].fit(
+    X_train_t,
+    y_train_t,
+    sample_weight=w_train_t,
+    eval_set=[(X_train_t, y_train_t), (X_test_t, y_test_t)],
+    eval_names=["train", "test"],
+    eval_metric="poisson",  # Scoring metric
+    callbacks=[lgb.record_evaluation(eval_results)]  # Record results for plotting
+)
+
+# Plot the learning curve
+lgb.plot_metric(eval_results, metric="poisson")
+plt.title("Learning Curve for LGBMRegressor")
+plt.ylabel("Poisson Deviance")
+plt.xlabel("Boosting Iterations")
+plt.show()
+
+
 
+# %%
+# PS4 EX3
+from ps3.evaluation import evaluate_predictions
```

Comment:
Can you put import change to the beginning of the file in the section where we group all the imports?